src/Controller/ImportController.php

```diff
@@ -0,0 +1,388 @@
+            ], Response::HTTP_BAD_REQUEST);
+        }
+        if ($token !== $this->nightlyToken) {
+            return new JsonResponse([
+                'message' => 'Invalid token',
+            ], Response::HTTP_UNAUTHORIZED);
+        }
+
+        preg_match('/[0-9]{4}-[0-9]{2}-[0-9]{2}-(.*)?\.json/', $this->filename, $matchesVersion);
+        if (!isset($matchesVersion[1])) {
+            return new JsonResponse([
+                'message' => 'Could not retrieve version from filename',
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        $this->version = $matchesVersion[1];
+        if (strlen($this->version) < 1) {
+            return new JsonResponse([
+                'message' => sprintf(
+                    'Version found not correct (%s) from filename %s',
+                    $this->version,
+                    $this->filename
+                ),
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        $fileContent = @file_get_contents($this->nightlyGCPUrl . 'reports/' . $this->filename);
+        if (!$fileContent) {
+            return new JsonResponse([
+                'message' => 'Unable to retrieve content from GCP URL',
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        $this->jsonContent = json_decode($fileContent);
+        if (!$this->jsonContent) {
+            return new JsonResponse([
+                'message' => 'Unable to decode JSON data',
+            ], Response::HTTP_BAD_REQUEST);
+        }
+
+        $force = $request->query->get('force', false);
+        $force = is_bool($force) ? $force : false;
+
+        $this->platform = $request->query->has('platform') ? $request->query->get('platform') : (
+            $request->query->has('browser') ? $request->query->get('browser') : null
+        );
+        $this->platform = in_array($this->platform, self::FILTER_PLATFORMS) ? $this->platform : self::FILTER_PLATFORMS[0];
+
+        $this->campaign = $request->query->has('campaign') ? $request->query->get('campaign') : null;
+        $this->campaign = in_array($this->campaign, self::FILTER_CAMPAIGNS) ? $this->campaign : self::FILTER_CAMPAIGNS[0];
+
+        $this->startDate = \DateTime::createFromFormat($dateFormat, $this->jsonContent->stats->start);
+
+        // Check if there is no similar entry
+        if (!$force && $this->executionRepository->findOneByNightly($this->version, $this->platform, $this->campaign, $this->startDate->format('Y-m-d'))) {
+            return new JsonResponse([
+                'message' => sprintf(
+                    'A similar entry was found (criteria: version %s, platform %s, campaign %s, date %s).',
+                    $this->version,
+                    $this->platform,
+                    $this->campaign,
+                    $this->startDate->format('Y-m-d')
+                ),
+            ], Response::HTTP_FORBIDDEN);
+        }
+
+        return null;
+    }
+
+    #[Route('/hook/add', methods: ['GET'])]
```

Comment:
Nope, it's a GET route : https://github.com/PrestaShop/QANightlyResults/blob/develop/src/routes.php#L28

Reply:
I know, that doesn't mean it was initially the right method That's what I was challenging since you are refactoring the tool, but maybe there are constraints that make it easier to keep the GET?

Reply:
That implies to change to each location (and each branch) where the API is called... So not at the moment. But I add it to my roadmap.